iothub/device/src/Authentication/IotHubConnectionCredentials.cs

```diff
@@ -26,23 +26,23 @@ internal IotHubConnectionCredentials()
         /// The authentication method that is used. It includes <see cref="ClientAuthenticationWithSharedAccessKeyRefresh"/>, <see cref="ClientAuthenticationWithSharedAccessSignature"/>,
         /// <see cref="ClientAuthenticationWithX509Certificate"/> or <see cref="EdgeModuleAuthenticationWithHsm"/>.
         /// </param>
-        /// <param name="iotHubHostName">The fully-qualified DNS host name of IoT hub.</param>
+        /// <param name="hostName">The fully-qualified DNS host name of IoT hub.</param>
```

Comment:
We're throwing an exception here based on this name, if null or whitespace. It bubbles up to the client ctor, though, which uses `hostName` as the param name.

We really shouldn't be doing this, because the caller doesn't know about `iotHubHostName` as a parameter as the API surface they used doesn't include that parameter.

Aside from fixing this callstack, I can just make the parameters the same. I don't love it, but I don't have time to do the full fix.